admin/failures/overview/endpoint.js

```diff
@@ -13,28 +13,23 @@ module.exports = (purchases) => (req, res) => {
         let failedOrders = orders
             .filter(x => x.status == "failed" || (x.data.errors && x.data.errors.length))
             .sort((a, b) => {
-                if(!a.data.errors || !a.data.errors.length) {
-                    return -1;
-                }
-                if(!b.data.errors || !b.data.errors.length) {
+                const aLatestTimestamp = getLatestTimestamp(a);
+                const bLatestTimestamp = getLatestTimestamp(b);
+
+                if(aLatestTimestamp < bLatestTimestamp) {
                     return 1;
                 }
-
-                const aLatest = a.data.errors[a.data.errors.length - 1];
-                const bLatest = b.data.errors[b.data.errors.length - 1];
-                if(aLatest.at < bLatest.at) {
+                if(aLatestTimestamp > bLatestTimestamp) {
                     return -1;
                 }
-                if(aLatest.at > bLatest.at) {
-                    return 1;
-                }
                 return 0;
             })
             .map((o) => {
-                const latestError = o.data.errors && o.data.errors.length && o.data.errors[o.data.errors.length - 1];
+                const latestError = getLastElement(o.data.errors);
+                const latestErrorAt = latestError ? prettifyDate(latestError.at) : prettifyDate(o.data.viewModel.date);
                 return {
                     id: o.id,
-                    latestErrorAt: latestError && prettifyDate(latestError.at),
+                    latestErrorAt,

```

Comment:
Instead of getting `latestError` and `latestErrorAt` you can now just use your nice function `getLatestTimestamp`:

```js
latestErrorAt: prettifyDate(getLatestTimestamp(o)),
```